src/lightning_lite/utilities/imports.py

```diff
@@ -19,7 +19,12 @@
 from lightning_utilities.core.imports import compare_version
 
 _IS_WINDOWS = platform.system() == "Windows"
-_IS_INTERACTIVE = hasattr(sys, "ps1")  # https://stackoverflow.com/a/64523765
+
+# There are two types of interactive mode we detect
+# 1. The interactive Python shell: https://stackoverflow.com/a/64523765
+# 2. The inspection mode via `python -i`: https://stackoverflow.com/a/6879085/1162383
+_IS_INTERACTIVE = hasattr(sys, "ps1") or bool(sys.flags.interactive)
```

Comment:
If anyone knows how to correctly mock `sys.flags.interactive`, please let me know. So far, I was not able to add a test.

Reply:
It seems that cpython is just directly mocking full `sys.flags` https://github.com/python/cpython/blob/286549b7cc8f3610014c3db204a693b4e219ef46/Lib/test/test_site.py#L344

Reply:
Thanks Ota. This is also what I tried. It looks like they can get away with this because they don't invoke any other Python code that needs these flags. However, for us, it seems that if we invoke things like `with pytest.raises` it calls Python code that needs to access `sys.flags` and freaks out about the Mock. I also tried collecting all existing attributes from `sys.flags` and wrapping them in mock without success.